student_explorer/cron.py

```diff
@@ -0,0 +1,191 @@
+            # The advisors of those department will be added as observers to student's Canvas courses
+            for dept in deptAffiliations:
+                if dept in user_affiliation:
+                    # advisor with right dept affiliation
+                    in_affiliated_department = True
+
+        return in_affiliated_department
+
+    def add_user_as_observer_to_course(self, class_site_id, mentor_canvas_user_id, user_canvas_user_id):
+        """
+        check whether the mentor is enrolled in this Canvas course
+        if enrolled, do nothing;
+        else add the mentor user to the Canvas course with Observer role
+        :param class_site_id:
+        :param mentor_canvas_user_id:
+        :return:
+        """
+        enrolled = False
+        course_section_id = ""
+        enrollments_resp = apiUtil.api_call(f"aa/CanvasReadOnly/courses/{class_site_id}/enrollments", "canvasreadonly")
+        if (enrollments_resp.text.find("errors") != -1):
+            # check for error message
+            # {"errors":[{"message":"The specified resource does not exist."}],"error_report_id":7008726002}
+            logger.warn(f"errors retrieving enrollments for class {class_site_id}: {enrollments_resp.text}")
+        else:
+            enrollments_json = json.loads(enrollments_resp.text)
+
+            for item in enrollments_json:
+                user_id=item.get("user_id")
+
+                # get the Canvas section id for student enrollment
+                if (user_id == user_canvas_user_id):
+                    course_section_id = item.get("course_section_id")
+
+                # check whether mentor has already enrolled in the course
+                if user_id == mentor_canvas_user_id:
+                    # advisor is enrolled in the course
+                    # break and do nothing
+                    enrolled = True
+                    break
+
+            logger.info(f" enrolled = {enrolled} user_canvas_user_id = {user_canvas_user_id} course_section_id={course_section_id}")
+            if enrolled:
+                # mentor is already in Canvas course site
+                # do nothing
+                logger.info(f"User id {mentor_canvas_user_id} is already enrolled in course {class_site_id}")
+            else:
+                #enroll mentor to the Canvas course site with Observer role
+                logger.info(f"Start: User id {mentor_canvas_user_id} is NOT enrolled in course {class_site_id}. Preparing to add user with Observer role. ")
+
+                # POST call to add mentor with Observer role to given section
+                payload_dictionary = {
+                    "enrollment[user_id]": mentor_canvas_user_id,
+                    "enrollment[type]": "ObserverEnrollment",
+                    "enrollment[enrollment_state]": "active",
+                    "enrollment[course_section_id]": course_section_id,
+                    "enrollment[limit_privileges_to_course_section]": "true",
```

Comment:
In Dan's example he wasn't checking this box. I don't know if checking it would have any effect or not. If they're associated with a specific user it's probably better to just leave it false.